v2/internal/carv1/car.go

```diff
@@ -209,3 +209,39 @@ func loadCarSlow(s Store, cr *CarReader) (*CarHeader, error) {
 		}
 	}
 }
+
+// Equals checks whether two headers are equal.
+// Two headers are considered equal if:
+//   1. They have the same version number, and
+//   2. They contain the same root CIDs in any order.
+func (h CarHeader) Equals(other CarHeader) bool {
```

Comment:
@masih Rename `Equal` to something else since `Equal` has a specific meaning in go, more in the lines of `reflect.DeepEqual` where order of elements matter.